### ISSUE TITLE
Support for Fedora by adding sesh to dnf package manager

### ISSUE THREAD

**Issue by rpajunen** [joshmedeski/sesh#181]
As the title says. I would love to use sesh in my fedora environment.

**Comment by joshmedeski** [comment]
GoReleaser doesn't specifically mention dnf or fedora, do you know if this would be sufficient?
https://goreleaser.com/customization/nfpm/

**Comment by rpajunen** [comment]
It seems like GoReleaser can be used to package to RPM via nfpm since DNF supports RPM.

**Comment by buckaroogeek** [comment]
As an FYI - I have Fedora rpms available in COPR: https://copr.fedorainfracloud.org/coprs/buckaroogeek/Tmux_sesh/

**Comment by TheSteinn** [comment]
Any update on this? 🙂

**Comment by buckaroogeek** [comment]
Still in COPR at the URL above.

**Comment by TheSteinn** [comment]
Cheers, yeah I did see that. Was just wondering if there had been any news on an official rpm. I usually stay away from copr when I can, though if it's not on the foreseeable roadmap I might have to bite the bullet this time

**Comment by buckaroogeek** [comment]
There is a build requirement for sesh called mockery that is stalled in the review process. I need to try and shake it loose. It will be several weeks before anything happens. I do keep the COPR current.

**Comment by joshmedeski** [comment]
@buckaroogeek what can I do to help? mockery is required for testing but not for building.

I've recently transitioned to a justfile so you can use the `just build` command if you need the binary.

But how does it work? I can update the github releases to have the proper binaries so you don't have to build it yourself. Unless Fedora requires building from source on releases.

**Comment by buckaroogeek** [comment]
@joshmedeski - Thanks for the offers. Fedora requires that official rpms build from upstream sources.  There is also an expectation that any relevant unit tests in the sources are also evaluated in the rpm %check phase. Hence the inclusion of mockery as a "build" requirement, i.e. build in the complete sense for the rpm.

**Comment by joshmedeski** [comment]
Yeah, all unit testing require mockery at this point. I used to build mockery mocks and put them in version control. But I found it was bloated and not necessary.

Perhaps we need to add them back to version control? Or create a step for fetching mockery before building for your pipeline?

**Comment by buckaroogeek** [comment]
Putting the mocks in version control is possible. For security purposes, the Fedora/RedHat rpm build environment blocks downloads. Another option could be something like a build tag for the unit tests that use mockery, and to just exclude that build tag during the testing phase. Or for the short term I could just note that unit tests are not viable until mockery becomes available in Fedora.

**Comment by joshmedeski** [comment]
On GitHub, every build has run unit tests to verify proper functionality.

I wouldn't want to create a separate tagging workflow just for one platform. Let me know what you think is the best next step.

Thanks for your help with this!

**Comment by buckaroogeek** [comment]
I will poke mockery and ask someone to review. Then sesh. Glad to help.

**Comment by buckaroogeek** [comment]
An update - I belatedly reached out to Landon with respect to the licensing problem I was seeing. He has a fix in the works (https://github.com/vektra/mockery/issues/1168#issuecomment-5108921874)

**Comment by buckaroogeek** [comment]
Mockery blocker resolved. Sesh v2.28.0 now building at https://copr.fedorainfracloud.org/coprs/buckaroogeek/Tmux_sesh/ for all Fedora versions (F43, F44, F45, rawhide) and EPEL10. The Fedora review process to add mockery as a package is restarted. Once mockery is accepted, I will submit sesh as a package for Fedora review and acceptance.

**Comment by joshmedeski** [comment]
That's great @buckaroogeek, thanks for doing this!

**Comment by joshmedeski** [comment]
I added it to the README. Can you confirm when it's deployed and working for you and I'll merge it?

**Comment by buckaroogeek** [comment]
Installed and working on my Fedora 44 workstation. The mockery rpm in now in review status. Once approved I will submit sesh for review.


<img width="981" height="1061" alt="Image" src="https://github.com/user-attachments/assets/dce62357-cbac-4f5f-9471-c91c88d0578c" />

**Comment by buckaroogeek** [comment]
mockery approved submitted to build environment. Should be in rawhide tomorrow (26 Aug).  Packages submitted for F45 (the forthcoming release) and F44 (current release). Mockery has go 1.26 set in the go.work file so will not be in F43 until go 1.26 is pushed in production next week.  Sesh package staged for review pending mockery reaching rawhide.

**Comment by joshmedeski** [comment]
Feel free to ping us more on this issue if you want to continue reporting progress on it. But since it's working as expected, I'm going to close this issue.
